### PR TITLE
SONARIAC-1093 Extend DockerLanguage to utilize filenamePatterns for non conflicting patterns

### PR DESCRIPTION
**Pull request by jonas-wielage-sonarsource**

### CODE REVIEW

**Comment by jonas-wielage-sonarsource** [comment]
Language Assignment Results were verified in SonarLint and SonarQube via UI / Debugger

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerSensorTest.java [hunk 1]
```diff
@@ -64,23 +68,59 @@ void shouldReturnActivationSettingKey() {
...
+      inputFileWithoutAssociatedLanguage("Foo.dockerfile", ""),
+      // should not be included after applying file predicates
+      inputFileWithoutAssociatedLanguage("DockerfileFoo", ""),
+      inputFileWithoutAssociatedLanguage("FooDockerfile", ""));
+
+    FileSystem fileSystem = sonarLintContext.fileSystem();
+    Iterable<InputFile> inputFiles = fileSystem.inputFiles(sonarLintSensor.mainFilePredicate(sonarLintContext));
+
+    assertThat(inputFiles)
+      .extracting(inputFile -> Path.of(inputFile.uri()).getFileName().toString())
```
Comment (petertrr) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerSensorTest.java:117]: `InputFile#filename` maybe?

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerLanguageTest.java [hunk 2]
```diff
@@ -19,15 +19,59 @@
...
+    "Dockerfile.foo"
+  })
+  void fileNameIsNotAssignedToLanguage(String fileName) {
+    assertThat(associatedToLanguage(fileName)).isFalse();
+  }
+
+  private static boolean associatedToLanguage(String fileName) {
+    DockerLanguage language = new DockerLanguage();
+
+    // Based on the actual implementation in SQ and SC
```
Comment (petertrr) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerLanguageTest.java:68]: Is it something from `sonar-plugin-api-impl`? I suggest to provide more details on where the implementation is to make it easier to adapt to potential changes in there

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/plugin/DockerSensor.java [hunk 3]
```diff
@@ -80,13 +80,21 @@ public void describe(SensorDescriptor descriptor) {
   protected FilePredicate mainFilePredicate(SensorContext sensorContext) {
     FileSystem fileSystem = sensorContext.fileSystem();
     FilePredicates p = fileSystem.predicates();
-    return p.and(
-      p.or(
+
+    FilePredicate pathPatterns = p.matchesPathPattern("**/Dockerfile.*");
+    if (!isNotSonarLintContext(sensorContext)) {
```
Comment (petertrr) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/plugin/DockerSensor.java:85]: Does this condition need a TODO to be removed after SonarLint adopts new API too?

**Review by petertrr — APPROVED** [review]
LGTM, just a couple of questions

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerLanguageTest.java (hunk 2 above)
Comment (jonas-wielage-sonarsource) [reply iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerLanguageTest.java:68]: No, its from sonarqube and sonarcloud directly, its in the `LanguageDetection` class. I will provide more details.

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerSensorTest.java (hunk 1 above)
Comment (jonas-wielage-sonarsource) [reply iac-extensions/docker/src/test/java/org/sonar/iac/docker/plugin/DockerSensorTest.java:117]: I shouldn't just copy paste from other tests :D

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/plugin/DockerSensor.java (hunk 3 above)
Comment (jonas-wielage-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/plugin/DockerSensor.java:85]: Added a reference to https://sonarsource.atlassian.net/browse/SLCORE-526

**Pull request merged**